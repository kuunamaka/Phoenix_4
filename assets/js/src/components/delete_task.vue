<template>
  <div><button @click="deleteTask(task)">Delete</button></div>
</template>

<script>
import axios from 'axios'
export default {
  name: 'delete_task',
  data() {
    return {
      task: ''
    }
  },
  methods: {
    async deleteTask(task) {
      try {
        const resp = await axios.delete(`/api/tasks` + `/${ task.id }`, {
          task: {
            name: this.name
          }
        })
        console.log(resp);
        window.location.href = '/tasks';
      }
      catch (e) {
        console.log(e)
      }
    }
  },
  async mounted() {
    try {
      const task_id = window.location.pathname.split('/')[2];
      const resp = await axios.get('/api' + `/tasks/${ task_id }`)
      this.task = resp.data
    }
    catch (e) {
      console.log(e)
    }
  }
}
</script>
