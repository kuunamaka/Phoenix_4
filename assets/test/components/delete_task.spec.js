import { shallowMount } from "@vue/test-utils";
import delete_task from "@/src/components/delete_task.vue";

const axios = require("axios");
const MockAdapter = require("axios-mock-adapter");
const mock = new MockAdapter(axios);

mock.onGet("/api/tasks/1").reply(200, {
  id: 1,
  name: "Test",
  is_finished: false,
  assignee_id: 1,
});

global.window = Object.create(window);
const pathname = "/tasks/1";

Object.defineProperty(window, "location", {
  value: {
    pathname,
    href: `http://localhost:4001${pathname}`,
  },
});

describe("delete_task.vue", () => {
  const wrapper = shallowMount(delete_task);
  const vm = wrapper.vm;

  it("renders entire component correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe("deleteTask()", () => {
    it("redirect to /tasks", async () => {
      mock
        .onDelete("/api/tasks/1")
        .reply(200, { task: { id: 1, name: "Test" } });
      await wrapper.setData({ task: { id: 1, name: "Test" } });

      await vm.deleteTask({ id: 1, name: "Test" });

      expect(window.location.href).toEqual("/tasks");
    });
  });
});
