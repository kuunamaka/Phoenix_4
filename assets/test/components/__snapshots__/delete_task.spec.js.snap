// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete_task.vue is a Vue instance 1`] = `
<div>
  <button>
    Delete
  </button>
</div>
`;
